### ISSUE TITLE
no versioning for css files

### ISSUE THREAD

**Issue by elasticsteve** [JeffreyWay/laravel-mix#211]
webpack.mix.js:

```
mix.js('resources/assets/js/pages/main.js', 'public/js/main.js')
    .js('resources/assets/js/pages/account.js', 'public/js')
    .sass('resources/assets/sass/pages/main.scss', 'public/css')
    .sourceMaps()
    .combine([
        'resources/assets/uikit/css/uikit-edit.css',
        'resources/assets/uikit/css/components/upload.gradient.css',
        'resources/assets/uikit/css/components/sticky.gradient.css',
        'node_modules/sweetalert/dist/sweetalert.css',
        'resources/assets/css/animate.css',
        'resources/assets/css/app.css'
    ], 'public/css/all.css')
    .version();
```

`public/css/all.css` does not get versioned. Only `.js` and `.sass`

![pic2](https://cloud.githubusercontent.com/assets/10718044/22393809/a4810c86-e506-11e6-90a2-8d497a5c28be.jpg)

manifest:
```
{
  "/js/account.js": "/js/account.ea71c09548b7227cbe14.js",
  "/js/main.js": "/js/main.8876eb3f57f68f278c2c.js",
  "/css/main.css": "/css/main.3f8e8514365bea403da7.css"
}
```

on Win 10

**Comment by maengkom** [comment]
I have same experience using combine().version() it will not generate versioning. Solution, I have to save combine() to another `resource/assets/generated/css/all.css` then using that file to versioning and save tu public folder.

**Comment by komirad** [comment]
Just had this problem as well. Would be nice to add this feature.

**Comment by unfortunatetoast** [comment]
+1

**Comment by r3k4** [comment]
+1

**Comment by adriaanzon** [comment]
This has been discussed before: #53, #154, #163.

Conclusion is that you can use `mix.sass()`, and import the css files in your sass entry point.

**Comment by elasticsteve** [comment]
@adriaanzon thanks.

You said in one of the threads:
> Using mix.sass for your css files directly seems to work too.

But when I do this:
```
mix.js('resources/assets/js/pages/main.js', 'public/js/main.js')
    .sass([
        'resources/assets/uikit/css/uikit-edit.css',
        'resources/assets/uikit/css/components/upload.gradient.css',
        'resources/assets/uikit/css/components/sticky.gradient.css',
        'node_modules/sweetalert/dist/sweetalert.css',
        'resources/assets/css/animate.css',
        'resources/assets/css/app.css'
    ], 'public/css/all.css')
    .version();
```

I get an error.
```
assert.js:85
  throw new assert.AssertionError({
  ^
AssertionError: mix.sass() is missing required parameter 1: src
```

**Comment by adriaanzon** [comment]
Yeah, you can't pass array to `mix.sass()`, only one file. So you'd have to use one sass file that imports the css files, something like this:

```javascript
// webpack.mix.js

mix.sass('resources/assets/sass/app.scss', 'public/css/all.css');
```

```scss
// resources/assets/sass/app.scss

@import "../uikit/css/uikit-edit.css";
@import "../uikit/css/components/upload.gradient.css";
@import "node_modules/sweetalert/dist/sweetalert.css";
// etc.
```

**Comment by elasticsteve** [comment]
I used this now, which works, but now `npm run watch` is stuck in an endless loop of compiling. It's compiling over and over again! How come?

```
mix.combine([
    'resources/assets/uikit/css/uikit-edit.css',
    'resources/assets/uikit/css/components/upload.gradient.css',
    'resources/assets/uikit/css/components/sticky.gradient.css',
    'node_modules/sweetalert/dist/sweetalert.css',
    'resources/assets/css/animate.css',
    'resources/assets/css/app.css'
], 'public/css/combine/all.css')
    .js('resources/assets/js/pages/main.js', 'public/js/main.js')
    .js('resources/assets/js/pages/account.js', 'public/js')
    .sass('public/css/combine/all.css', 'public/css/all.css')
    .sass('resources/assets/sass/pages/main.scss', 'public/css')
    .sourceMaps()
    .version();
```

**Comment by adriaanzon** [comment]
Looks like you're not the only one: #228

**Comment by elasticsteve** [comment]
@adriaanzon Superb! Of course! Why not import all the css in the app.scss! Very nice.

I think Taylor should add a little note to the docs, because this will trip people up over and over until they drive @JeffreyWay nuts with the same question. haha

**Comment by JeffreyWay** [comment]
Yeah the thing to understand is that `mix.copy()` and `mix.combine()` are extras that aren't part of the Webpack compile process. Think of them as little hooks that get triggered once Webpack is done compiling. "When you finish compiling, combine these files." 

They're not part of the core bundling process, so they won't be versioned. If they need to be, introduce the files into your mix.js() or mix.sass() calls.

**Comment by nejtr0n** [comment]
So, it's not compataible with elixir version on styles
```
        //Concat all scripts in one
        .scripts(
            modules
                .map(function(module) {
                    return module.paths.admin_js;
                })
                .filter(filter_assets)
            , 'public/js/admin/all.js')
        .version([
            'public/css/admin/all.css',
            'public/js/admin/all.js'
        ])
```
Please return elixr!

**Comment by elasticsteve** [comment]
It's a different product. You can get it all done, but have to change your approach. You can still use Elixir if you like. I felt like you first, but now I prefer Mix (when it works - there were/are some issues on Windows).

**Comment by jonathanpmartins** [comment]
With ``mix.js()`` I get a different result then ``mix.combine()``. Using ``mix.js`` Im also getting errors, because I think its trying to analyse and transpile it.

Think I will learn more about webpack and mix, so hopefully soon we all can get it done.

**Comment by JeffreyWay** [comment]
mix.js and mix.combine do different things, so it makes sense that you'll get a different result. 

What error are you getting?

**Comment by Icesik** [comment]
Not sure I got well your issue, but I found this thread through my research for a similar problem, and the solution I've found may help someone.

The problem is that css files generated from sass compilation are versioned, so they have a different name from original, just use wildcard for styles() or combine() use.

```
mix
.sass('resources/assets/sass/app.scss', 'public/css/app.css')
.styles([
        'public/css/app.*.css',
        ...
])
.version()
;
```

Hope it helps!
